Software/api/src/main/java/org/foi/air/api/request_handlers/LoginRequestHandler.kt

```diff
@@ -0,0 +1,64 @@
+package org.foi.air.api.request_handlers
+
+import ResponseListener
+import com.google.gson.Gson
+import okhttp3.ResponseBody
+import org.foi.air.api.models.LoginBody
+import org.foi.air.api.network.ApiService
+import org.foi.air.core.data_classes.UserInfo
+import org.foi.air.core.network.RequestHandler
+import org.foi.air.core.models.ErrorResponseBody
+import org.foi.air.core.models.SuccessfulLoginResponseBody
+import org.json.JSONObject
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class LoginRequestHandler(private val requestBody: LoginBody): RequestHandler<SuccessfulLoginResponseBody> {
+    override fun sendRequest(responseListener: ResponseListener<SuccessfulLoginResponseBody>){
+        val service = ApiService.authService
+        val serviceCall = service.loginUser(requestBody)
+
+        serviceCall.enqueue(object : Callback<ResponseBody>{
+            override fun onResponse(call: Call<ResponseBody>, response: Response<ResponseBody>) {
+                if(response.isSuccessful){
+                    val convertedResponse = successfulResponseConverter(response.body()!!)
+                    if(convertedResponse == null) {
+                        responseListener.onErrorResponse(ErrorResponseBody(false,"Failed to convert request and login user","Unsuccessful request conversion"))
```

Comment:
I don't think the error message should include "Failed to convert request", the user has no use for that information.

Reply:
Thank you for the insight. I changed the messages accordingly.